### ISSUE TITLE
Add suport for build artifacts in pipelines.

### ISSUE THREAD

**Issue by kyngs** [woodpecker-ci/woodpecker#1577]
### Clear and concise description of the problem

I have a pipeline which builds my project. I want to attach the JAR files it builds to the pipeline.

### Suggested solution

It would work similarly to Jenkins build artifacts.
Perhaps in this section:
![image](https://user-images.githubusercontent.com/38181667/216778398-b81f72de-09e4-4c16-bea9-e380d7242c40.png)
a tab named something like "Build Artifacts" could be added.
The tab should contain the files, which the viewer could download.

### Alternative

Adding a step that would upload the artifacts to an external server.

### Additional context

I couldn't find any issues on the same subjects, which seems surprising considering that this is a pretty popular use case.
Maybe someone already suggested it, and it was rejected? I don't know.

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by 6543** [comment]
Must be optional and should only be available if an s3 storage option was provided to woodpecker

**Comment by 6543** [comment]
@kyngs you already can upload artifacts as you want in any way but it's not shown in woodpecker itself ...

**Comment by kyngs** [comment]
> Must be optional and should only be available if an s3 storage option was provided to woodpecker

My original idea was, that it would be stored on the woodpecker itself or some other self-hosted solution.

**Comment by neel** [comment]
@kyngs gitlab-runner has artifacts. I was looking for something like that in Woodpecker. It doesn't require S3. Are you also requesting for similar feature ?

**Comment by kyngs** [comment]
> @kyngs gitlab-runner has artifacts. I was looking for something like that in Woodpecker. It doesn't require S3. Are you also requesting for similar feature ?

Yes, Jenkins also has this feature, and so do Github actions. It is really useful for sharing binaries for each commit.

**Comment by neel** [comment]
> > @kyngs gitlab-runner has artifacts. I was looking for something like that in Woodpecker. It doesn't require S3. Are you also requesting for similar feature ?
> 
> Yes, Jenkins also has this feature, and so do Github actions. It is really useful for sharing binaries for each commit.

Not only that, it is very useful for writing papers in Latex. For every push the pdf is generated automatically.
